libpromises/evalfunction.c

```diff
@@ -2733,10 +2756,29 @@ static FnCallResult FnCallDatastate(EvalContext *ctx,
                                     ARG_UNUSED const FnCall *fp,
                                     ARG_UNUSED const Rlist *args)
 {
-    JsonElement *state = DefaultTemplateData(ctx);
+    JsonElement *state = DefaultTemplateData(ctx, NULL);
     return  (FnCallResult) { FNCALL_SUCCESS, (Rval) { state, RVAL_TYPE_CONTAINER } };
 }
 
+static FnCallResult FnCallBundlestate(EvalContext *ctx,
+                                      ARG_UNUSED const Policy *policy,
+                                      ARG_UNUSED const FnCall *fp,
+                                      ARG_UNUSED const Rlist *args)
+{
+    JsonElement *state = DefaultTemplateData(ctx, RlistScalarValue(args));
+
+    if (NULL == state ||
+        JsonGetElementType(state) != JSON_ELEMENT_TYPE_CONTAINER ||
+        JsonLength(state) < 1)
+    {
+        return FnFailure();
```

Comment:
Very corner case, I know, but perhaps it should distinguish between a failure to find the bundle and a bundle that's empty. With this code you would get an unexpanded data container if you tried to use it after returning data from an empty bundle.

That is, unless there is some bundle member that is always returned.